data_validation/secret_manager.py

```diff
@@ -50,5 +53,6 @@ def maybe_secret(self, project_id, secret_id, version_id="latest"):
             # Return the decoded payload.
             payload = response.payload.data.decode("UTF-8")
             return payload
-        except BaseException:
+        except exceptions.NotFound:
```

Comment:
The main reason I didn't create a PR previously is the risk of this line.

I think it is better to tell the user if they have a real problem and therefore I think this change is good. My only concern is that if we have customers who have incorrectly included the secret project id in their connection files then it will currently be silently failing. But after this change the exception will bubble to the top.

On balance I still think this code is better than before. But we should expect someone somewhere to see an exception they did not know they were hitting.

Reply:
Agreed! I wonder if it would be the case of marking it as a breaking change? But since it's something not in the main and critical path, maybe it's not worth it...

Reply:
Yeah, I think it's not worth marking as a breaking change. Just something for us to be aware of.